proposals/0293-extend-property-wrappers-to-function-and-closure-parameters.md

```diff
@@ -180,81 +145,116 @@ struct Traceable<Value> {
+
+The natural model for these two kinds of wrappers is different when applied to parameters, because the second model must allow the caller to pass a different type of argument. We propose to formalize the difference between API-level property wrappers that have an external effect on the function and implementation-detail property wrappers. The compiler will determine whether a property wrapper must have an external effect on the function by analyzing the property wrapper's initializers.
 
-Using property-wrapper parameters, the above argument validation example can be simplified to:
+Annotating a parameter declaration with a property-wrapper attribute allows the call-site to pass a wrapped value, or a projected value if supported by the wrapper, and the compiler will automatically initialize the backing wrapper. The function author can also use the property-wrapper syntax for accessing the backing wrapper and the projected value within the body of the function.
+
+## Detailed design
+
+### Inference of API-level property wrappers
+
+For a given property wrapper attached to a parameter, the compiler will determine whether that wrapper is part of the function signature based on whether the wrapper must have an external effect on the argument at the call-site. There are two cases where the property wrapper must affect the argument at the call-site:
+
+1. The property wrapper supports projected-value initialization, allowing the caller to pass an instance of the projected-value type.
+2. The property wrapper accepts an `@autoclosure` to `init(wrappedValue:)`.
+
+To ensure that the compiler always makes the same decision regardless of which module the property wrapper is used from, the compiler will only look in the defining module of the property wrapper for these initializers. To account for overloading and property wrapper composition, the compiler will infer the complete property wrapper [interface type](https://github.com/apple/swift/blob/main/docs/Lexicon.md#interface-type) before determining whether the property wrapper has an external effect.
+
+### Implementation-detail property wrappers
+
+By default, property wrappers are implementation detail. Attaching an implementation-detail property wrapper attribute to a parameter is sugar for declaring a local variable that's initialized via wrapped value using the parameter. Consider the following code, which attaches the `@Logged` property wrapper to a parameter.
 
 ```swift
-func buy(
-  @Asserted(.greaterOrEqual(1)) quantity: Int,
-  of product: Product,
-) {
-  if quantity == 1 {
-    ...
-  }
+func insert(@Logged text: String) { ... }
+```
+
+The above code is sugar for:
+
+```swift
+func insert(text: String) {
+  @Logged var text = text
```

Comment:
Perhaps we should avoid describing the transformation of implementation-detail wrappers as that of local property wrappers, due to their different setter semantics. Instead, we could refer to the transformation as local-wrapper-like (or something to that effect), if we don't remove such references entirely.

Reply:
I totally removed the term "local property wrapper" and went straight to the full desugared code. Let me know what you think or if it needs more clarification